src/views/virtualmachines/details/tabs/details/components/CPUMemory/CPUMemory.tsx

```diff
@@ -1,21 +1,40 @@
-import * as React from 'react';
+import React, { FC } from 'react';
 
-import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import {
+  V1alpha2VirtualMachineInstancetype,
+  V1VirtualMachine,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
 import { vCPUCount } from '@kubevirt-utils/resources/template/utils';
 import { readableSizeUnit } from '@kubevirt-utils/utils/units';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { Skeleton } from '@patternfly/react-core';
 
 type CPUMemoryProps = {
   vm: V1VirtualMachine;
 };
 
-const CPUMemory: React.FC<CPUMemoryProps> = ({ vm }) => {
+const CPUMemory: FC<CPUMemoryProps> = ({ vm }) => {
   const { t } = useKubevirtTranslation();
 
-  const cpu = vCPUCount(vm?.spec?.template?.spec?.domain?.cpu);
+  const [instanceType, loaded] = useK8sWatchResource<V1alpha2VirtualMachineInstancetype>(
+    vm?.spec?.instancetype?.name && {
+      groupVersionKind: {
+        group: 'instancetype.kubevirt.io',
+        version: 'v1alpha2',
+        kind: 'VirtualMachineInstancetype',
+      },
+      name: vm?.spec?.instancetype?.name,
+    },
+  );
+
+  if (!vm || !loaded) return <Skeleton />;
+
+  const cpu = instanceType?.spec?.cpu?.guest || vCPUCount(vm?.spec?.template?.spec?.domain?.cpu);
 
   const memory = readableSizeUnit(
-    (vm?.spec?.template?.spec?.domain?.resources?.requests as { [key: string]: string })?.memory,
+    (vm?.spec?.template?.spec?.domain?.resources?.requests as { [key: string]: string })?.memory ||
+      instanceType?.spec?.memory?.guest,
```

Comment:
I am not sure about this. Let's look at `cpu` variable just a few lines above: we look at `instanceType` first for cpu, then at `vm`. But here, when setting `memory`, we are doing it the opposite way. I think we better should be consistent in this.  I can imagine we can have both defined and get confusing results because of that. Please, take a look at this and choose what has a higher priority to look at for the values :)

Reply:
we can't have both at the same time

Reply:
That's good. And I'd keep the consistency anyway. It helps better readability of the code.

Reply:
fixed